src/Lumi/Components/Form.purs

```diff
@@ -383,6 +403,92 @@ switch = formBuilder_ \{ readonly } b onChange ->
            , onChange = Events.handler (stopPropagation >>> targetChecked) (traverse_ onChange)
            }
 
+-- | A `checkbox` is an editor for booleans which displays checked or not checked.
+checkbox :: forall props. FormBuilder { readonly :: Boolean | props } Boolean Boolean
+checkbox = labeledCheckbox mempty
+
+-- | A `labeledCheckbox` is an editor that behaves exactly like `checkbox` but
+-- | also accepts a JSX displayed as a label to its right.
+labeledCheckbox
+  :: forall props
+  . JSX
+  -> FormBuilder { readonly :: Boolean | props } Boolean Boolean
+labeledCheckbox label =
+  formBuilder_ \{ readonly } value onChange ->
+    Input.label
+      { style: R.css
+          { flexDirection: "row"
+          , alignSelf: "stretch"
+          , alignItems: "baseline"
+          }
+      , for: null
+      , children:
+          [ Input.input Input.checkbox
+              { style = R.css { marginBottom: "0", alignSelf: "baseline" }
+              , checked = if value then Input.On else Input.Off
+              , disabled = if readonly then true else false
+              , onChange =
+                  if readonly
+                    then Events.handler Events.syntheticEvent \_ -> pure unit
+                    else Events.handler (stopPropagation >>> targetChecked) (traverse_ onChange)
+              }
+          , lumiAlignToInput
+              { style: { flex: "1", marginLeft: "8px" }
+              , children: label
+              }
+          ]
+      }
+  where
+    lumiAlignToInput = element (R.unsafeCreateDOMComponent "lumi-align-to-input")
+-- checkbox
```

Comment:
Were you meaning to remove this?

Reply:
Oh yes sorry just had commented out for testing, will remove.